### ISSUE TITLE
Game crash after remove from recent apps

### ISSUE THREAD

**Issue by emmandyar** [reloZid/android-anuto#147]
The game FC's after removing it's "window" from recent apps menu. Here is the log as reported by my system (xiaomi redmi note 4, Miui 9.6.3.0 global)

```
java.lang.RuntimeException: Error in game loop!
	at ch.logixisland.anuto.engine.logic.loop.GameLoop.run(GameLoop.java:121)
	at java.lang.Thread.run(Thread.java:760)
Caused by: java.lang.ClassCastException: ch.logixisland.anuto.entity.shot.Mine cannot be cast to ch.logixisland.anuto.entity.plateau.Plateau
	at ch.logixisland.anuto.entity.tower.TowerPersister.readEntityData(TowerPersister.java:49)
	at ch.logixisland.anuto.entity.tower.TowerPersister.readEntityData(TowerPersister.java:10)
	at ch.logixisland.anuto.engine.logic.entity.EntityPersister.readState(EntityPersister.java:44)
	at ch.logixisland.anuto.engine.logic.persistence.GamePersister.readState(GamePersister.java:24)
	at ch.logixisland.anuto.business.game.GameLoader.initializeGame(GameLoader.java:191)
	at ch.logixisland.anuto.business.game.GameLoader.loadGame(GameLoader.java:148)
	at ch.logixisland.anuto.business.game.GameLoader$3.execute(GameLoader.java:119)
	at ch.logixisland.anuto.engine.logic.loop.MessageQueue.processMessages(MessageQueue.java:55)
	at ch.logixisland.anuto.engine.logic.loop.GameLoop.executeCycle(GameLoop.java:129)
	at ch.logixisland.anuto.engine.logic.loop.GameLoop.run(GameLoop.java:94)
	... 1 more
```

**Comment by reloZid** [comment]
I have trouble reproducing this save/load game issue. Could you give some more information? What is the state of the game before you close it (screenshot)? I assume the crash happens if you open the game again, but only once. Is that correct? Do you remove it from the recent apps directly after you closed the game?

**Comment by emmandyar** [comment]
Hi!

I managed to make it crash again, this is the bug report that miui gives me:

java.lang.RuntimeException: Error in game loop!
at ch.logixisland.anuto.engine.logic.loop.GameLoop.run(GameLoop.java:121)
at java.lang.Thread.run(Thread.java:760)
Caused by: java.lang.ClassCastException:
ch.logixisland.anuto.entity.shot.Mine cannot be cast to
ch.logixisland.anuto.entity.plateau.Plateau
at
ch.logixisland.anuto.entity.tower.TowerPersister.readEntityData(TowerPersister.java:49)
at
ch.logixisland.anuto.entity.tower.TowerPersister.readEntityData(TowerPersister.java:10)
at
ch.logixisland.anuto.engine.logic.entity.EntityPersister.readState(EntityPersister.java:44)
at
ch.logixisland.anuto.engine.logic.persistence.GamePersister.readState(GamePersister.java:24)
at
ch.logixisland.anuto.business.game.GameLoader.initializeGame(GameLoader.java:191)
at
ch.logixisland.anuto.business.game.GameLoader.loadGame(GameLoader.java:148)
at
ch.logixisland.anuto.business.game.GameLoader$3.execute(GameLoader.java:119)
at
ch.logixisland.anuto.engine.logic.loop.MessageQueue.processMessages(MessageQueue.java:55)
at
ch.logixisland.anuto.engine.logic.loop.GameLoop.executeCycle(GameLoop.java:129)
at ch.logixisland.anuto.engine.logic.loop.GameLoop.run(GameLoop.java:94)
... 1 more

**Comment by emmandyar** [comment]
Ok, I managed to take a screenshot before the crash: https://i.imgur.com/wiNpkcP.png

**Comment by reloZid** [comment]
Thank you so much! This actually allowed me to reproduce the issue. I fixed it in v0.5 (currently on beta).
